pkg/assembler/backends/helper/preloads.go

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helper
+
+import (
+	"context"
+
+	"github.com/99designs/gqlgen/graphql"
+)
+
+// getPreloads get the specific graphQL query fields that are requested.
+// graphql.CollectAllFields only provides the top level fields and none of the nested fields below it.
+// getPreloads recursively goes through the fields and retrieves each nested field below it.
+// for example:
+/*
+  type
+  namespaces {
+    namespace
+    names {
+      name
+      tag
+      commit
+    }
+  }
+  will return:
+  fields: [type namespaces namespaces.namespace namespaces.names namespaces.names.name namespaces.names.tag namespaces.names.commit]
+*/
+func GetPreloads(ctx context.Context) []string {
```

Comment:
how is this @mihaimaruseac ?

Reply:
ugh this causes the tests to fail:

https://github.com/99designs/gqlgen/blob/f9eac5c040f084424d13ea2edc068c0cda46da9c/graphql/context_operation.go#L61-L66


any way to get around this?

Reply:
This is it!

Hmm, for the test failure, I think the link is pointing to something else?

Reply:
cool this is fixed! Thanks @mihaimaruseac!